### PR TITLE
Add gcc11 to Ubuntu20.04 setup and add PkgConfig install

### PR DESCRIPTION
**Pull request by czentgr**
Modernize the compiler used on Ubuntu 20.04. Once Velox moves to using the C++20 standard the system gcc9 compiler cannot be used anymore.

In addition, recent upgrades added the usage of PkgConfig which was not installed into the system and is not by default pre-installed into ubuntu 20.04.

Resolves: https://github.com/facebookincubator/velox/issues/10953

### CODE REVIEW

**Comment by czentgr** [comment]
@PHILO-HE FYI.

**Comment by PHILO-HE** [comment]
@czentgr, the recently upgraded folly requires GCC >= 10. Otherwise, the below compile error will be reported. So we should land this pr now to avoid this error, right?
```
/velox/deps-download/folly/folly/Portability.h:38:24: error: static assertion failed: __GNUC__ >= 10
   38 | static_assert(__GNUC__ >= 10, "__GNUC__ >= 10");
```

**Comment by czentgr** [comment]
@PHILO-HE Yes, basically, Ubuntu 20.04 is currently broken without these changes - the user would do all of this manually.
We already state `Velox requires a compiler at the minimum GCC 11.0 or Clang 15.0.` in the README.

**Comment by czentgr** [comment]
@majetideepak @assignUser Please take a look.

File: scripts/setup-ubuntu.sh [hunk 1]
```diff
@@ -75,14 +91,19 @@ function install_build_prerequisites {
     ninja-build \
     checkinstall \
     git \
+    pkg-config \
+    cmake-data \
```
Comment (assignUser) [inline scripts/setup-ubuntu.sh:95]: nit:)Do we actually need cmake-data when installing via pip?

**Review by assignUser — APPROVED** [review]
Thanks!

File: scripts/setup-ubuntu.sh (hunk 1 above)
Comment (czentgr) [reply scripts/setup-ubuntu.sh:95]: Good point. We don't install cmake from the system. When looking for PkgConfig I though we might need more stuff. But this doesn't make sense. I'm trying it out without this package.

**Comment by PHILO-HE** [comment]
The pr looks good to me. Thanks!

**Comment by PHILO-HE** [comment]
@xiaoxmeng, if no comment, could you merge this pr? I just found this pr is also required for ubuntu 22.04 to fix missing pkg-config.

**Comment by mbasmanova** [comment]
@czentgr Would you rebase to allow merging?

**Comment by czentgr** [comment]
@mbasmanova I rebased. Please take another look.

**Pull request closed without merging**